lfs.c

```diff
@@ -480,7 +475,7 @@ static int lfs_alloc(lfs_t *lfs, lfs_block_t *block) {
 
         // find mask of free blocks from tree
         memset(lfs->free.buffer, 0, lfs->cfg->lookahead_size);
-        int err = lfs_fs_traverse(lfs, lfs_alloc_lookahead, lfs);
+        int err = lfs_fs_traverseraw(lfs, lfs_alloc_lookahead, lfs, true);
         if (err) {
             return err;
```

Comment:
Should reset .i, .size, .off incase of err, we want to make sure that we force a traverse next time lfs_alloc is called. 
I'm going to make a PR related to https://github.com/ARMmbed/littlefs/issues/394 but may need help updating the testing suite as I'm not certain the best way to make that change.